### PR TITLE
Fix MimeTypeUtil to not throw NPE when the content type can't be gues…

### PR DESCRIPTION
**Pull request by anuraaga**
…sed.

### CODE REVIEW

**Comment by anuraaga** [comment]
@trustin PTAL

File: src/main/java/com/linecorp/armeria/server/http/file/MimeTypeUtil.java [hunk 1]
```diff
@@ -88,7 +91,16 @@ static MediaType guessFromPath(String path) {
 
         final String extension = path.substring(dotIdx + 1).toLowerCase(Locale.US);
         final MediaType mediaType = EXTENSION_TO_MEDIA_TYPE.get(extension);
-        return mediaType != null ? mediaType : MediaType.parse(URLConnection.guessContentTypeFromName(path));
+        if (mediaType != null) {
+            return mediaType;
+        }
+        final String guessedContentType;
+        try {
+             guessedContentType = Files.probeContentType(Paths.get(path));
```
Comment (anuraaga) [inline src/main/java/com/linecorp/armeria/server/http/file/MimeTypeUtil.java]: Since the unit tests are passing with all fake files, I guess not.

Comment (trustin) [inline src/main/java/com/linecorp/armeria/server/http/file/MimeTypeUtil.java]: It passes at the moment, but it involves service loaders which means a user can insert any probing implementations.

Comment (anuraaga) [inline src/main/java/com/linecorp/armeria/server/http/file/MimeTypeUtil.java]: Hmm - ok, switched back to old version

**Comment by trustin** [comment]
Thanks, @anuraaga :+1:

**Pull request merged**